### PR TITLE
feat(CapMan): Tagstore Org IDs passed to Snuba

### PR DESCRIPTION
**Pull request by rahul-kumar-saini**
- All Tagstore Snuba queries now pass the `organization_id` tenant_id in the request
- See https://github.com/getsentry/sentry/pull/44788 for more context

### CODE REVIEW

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -811,16 +811,25 @@ def __seen_stats_impl(
 
         project_id = issue_list[0].project_id
         item_ids = [g.id for g in issue_list]
+        tenant_ids = {"organization_id": issue_list[0].project.organization_id}
```
Comment (wedamija) [inline src/sentry/api/serializers/models/group.py:814]: Just noting that we preload group projects here https://github.com/getsentry/sentry/blob/c8bbb69c609b5537d03f3da8c0aafda338e89293/src/sentry/api/serializers/models/group.py#L214, so this won't cause an extra query.

**Review by wedamija — APPROVED** [review]
lgtm, although looks like you need to add tenant ids to `sentry.snuba.metrics.get_tag_values`

File: src/sentry/tagstore/snuba/backend.py [hunk 2]
```diff
@@ -1489,3 +1582,7 @@ def get_group_event_filter(self, project_id, group_id, environment_ids, tags, st
             return None
 
         return {"event_id__in": event_id_set}
+        if not event_id_set:
```
Comment (evanh) [inline src/sentry/tagstore/snuba/backend.py:1585]: Why did you add this?

Comment (rahul-kumar-saini) [reply src/sentry/tagstore/snuba/backend.py:1585]: Must be something duplicated on a merge, wasn't intentional

File: src/sentry/api/serializers/models/group.py (hunk 1 above)
Comment (rahul-kumar-saini) [reply src/sentry/api/serializers/models/group.py:814]: I'm new to Django, are you saying I'm causing an extra query by doing `.project.organization_id`? How can I avoid this if so?

**Comment by rahul-kumar-saini** [comment]
> lgtm, although looks like you need to add tenant ids to `sentry.snuba.metrics.get_tag_values`

[`sentry.snuba.metrics.get_tag_values`](https://github.com/getsentry/sentry/blob/5bea27c8f7efaff0afc08320a78dcdb52c2df505/src/sentry/snuba/metrics/datasource.py#L479) calls [`_fetch...`](https://github.com/getsentry/sentry/blob/5bea27c8f7efaff0afc08320a78dcdb52c2df505/src/sentry/snuba/metrics/datasource.py#L297) which calls [`run_metrics_query`](https://github.com/getsentry/sentry/blob/183f9f2d22cd7f9aa409819b3a05bd3f3a1a6291/src/sentry/snuba/metrics/fields/base.py#LL113C15-L113C15) which contains the [`tenant_ids`](https://github.com/getsentry/sentry/blob/183f9f2d22cd7f9aa409819b3a05bd3f3a1a6291/src/sentry/snuba/metrics/fields/base.py#L152) in the Request.

Lmk if I'm missing any other query somewhere in this stack 😅

File: src/sentry/api/serializers/models/group.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/serializers/models/group.py:814]: No, just noting for anyone else reading that it won't cause any extra load

**Comment by wedamija** [comment]
> > lgtm, although looks like you need to add tenant ids to `sentry.snuba.metrics.get_tag_values`
> 
> [`sentry.snuba.metrics.get_tag_values`](https://github.com/getsentry/sentry/blob/5bea27c8f7efaff0afc08320a78dcdb52c2df505/src/sentry/snuba/metrics/datasource.py#L479) calls [`_fetch...`](https://github.com/getsentry/sentry/blob/5bea27c8f7efaff0afc08320a78dcdb52c2df505/src/sentry/snuba/metrics/datasource.py#L297) which calls [`run_metrics_query`](https://github.com/getsentry/sentry/blob/183f9f2d22cd7f9aa409819b3a05bd3f3a1a6291/src/sentry/snuba/metrics/fields/base.py#LL113C15-L113C15) which contains the [`tenant_ids`](https://github.com/getsentry/sentry/blob/183f9f2d22cd7f9aa409819b3a05bd3f3a1a6291/src/sentry/snuba/metrics/fields/base.py#L152) in the Request.
> 
> Lmk if I'm missing any other query somewhere in this stack 😅

This is what's causing the test failure though - something is attempting to pass tenant_ids to `sentry.snuba.metrics.get_tag_values` and it's not part of the function definition

**Review by wedamija — APPROVED** [review]
